lighthouse-core/gather/gatherers/url.js

```diff
@@ -21,7 +21,11 @@ const Gatherer = require('./gatherer');
 class URL extends Gatherer {
 
   beforePass(options) {
-    this.artifact = options.url || options.driver.url;
+    // Used currently by cache-start-url audit, which wants to know if the start_url
+    // in the manifest is stored in the cache.
+    // Instead of the originally inputted URL (options.initialUrl), we want the resolved
+    // post-redirect URL (which is here at options.url)
+    this.artifact = options.url;
```

Comment:
wouldn't you want this `afterPass`, otherwise redirection wouldn't have happened yet?